.github/workflows/lint.yml

```diff
@@ -0,0 +1,23 @@
+name: Lint on pull requests (or manually)
```

Comment:
we agreed to name things, so we all could use this repository for source of templates. 

We do have linting in other languages. We do have other languages than Go. 

I believe we talked about some nomenclature to make us more comfortable here.  ex: `[language]-[action].yaml` for instance.